html_parser.js

```diff
@@ -91,7 +91,7 @@ function add_courses_to_take(json, lines, j) {
     for(var i = 0; i < courseList.length; i++) {
         var course = { };
         if(!(courseList[i] == ' ' || courseList[i] == '' || courseList[i] == '****')) {
-            course.attr = courseList[i].substring(0, 4).split(' ').join('');
+            course.course.subject = courseList[i].substring(0, 4).split(' ').join('');
```

Comment:
I think this should be course.subject instead of course.course.subject

Reply:
> Additionally, [here](https://github.com/sandboxneu/scheduleneu/blob/9b27a1cbcd210ff7f62f25dd2087fb75e741db4a/html_parser.js#L144) (lines 143 and 144), season and year should be replaced with a new metric of when classes were taken, "termId", which looks something like "201830" (in String format) meaning 2018 => 2018, 30 => spring.
> 
> Also note that each termId starts in the fall of the previous year. For example the termIds starting with 2018:
> 
> * 201810 = fall 2017
> * 201830 = spring 2018
> * 201840 = summerI 2018
> * 201850 = summer full 2018
> * 201860 = summerII 2018
Additionally, [here](https://github.com/sandboxneu/scheduleneu/blob/9b27a1cbcd210ff7f62f25dd2087fb75e741db4a/html_parser.js#L144) (lines 143 and 144), season and year should be replaced with a new metric of when classes were taken, "termId", which looks something like "201830" (in String format) meaning 2018 => 2018, 30 => spring.

Also note that each termId starts in the fall of the previous year. For example the termIds starting with 2018:

* 201810 = fall 2017
* 201830 = spring 2018
* 201840 = summerI 2018
* 201850 = summer full 2018
* 201860 = summerII 2018

Got it - I had lopped off the two numbers at the end as I thought the year would be easier to work with. Good catch.